pkg/feature/manifest.go

```diff
@@ -1,76 +1,103 @@
 package feature
 
 import (
+	"bytes"
+	"embed"
 	"fmt"
 	"html/template"
-	"os"
+	"io"
+	"io/fs"
 	"path/filepath"
 	"strings"
-
-	"github.com/pkg/errors"
 )
 
-const BaseOutputDir = "/tmp/odh-operator"
+//go:embed templates
+var embeddedFiles embed.FS
 
 type manifest struct {
 	name,
-	path string
+	path,
+	processedContent string
 	template,
-	patch,
-	processed bool
+	patch bool
 }
 
-func loadManifestsFrom(path string) ([]manifest, error) {
+func loadManifestsFrom(fsys fs.FS, path string) ([]manifest, error) {
 	var manifests []manifest
-	if err := filepath.Walk(path, func(path string, info os.FileInfo, err error) error {
+
+	err := fs.WalkDir(fsys, path, func(path string, dirEntry fs.DirEntry, err error) error {
 		if err != nil {
 			return err
 		}
-		if info.IsDir() {
+
+		if dirEntry.IsDir() {
 			return nil
 		}
-		basePath := filepath.Base(path)
-		manifests = append(manifests, manifest{
-			name:     basePath,
-			path:     path,
-			patch:    strings.Contains(basePath, ".patch"),
-			template: filepath.Ext(path) == ".tmpl",
-		})
+
+		_, err = fs.ReadFile(fsys, path)
+		if err != nil {
+			return err
+		}
+		m := loadManifestFrom(path)
+		manifests = append(manifests, m)
 
 		return nil
-	}); err != nil {
-		return nil, errors.WithStack(err)
+	})
+
+	if err != nil {
+		return nil, err
 	}
 
 	return manifests, nil
 }
 
+func loadManifestFrom(path string) manifest {
+	basePath := filepath.Base(path)
+	m := manifest{
+		name:     basePath,
+		path:     path,
+		patch:    strings.Contains(basePath, ".patch"),
+		template: filepath.Ext(path) == ".tmpl",
+	}
+
+	return m
+}
+
 func (m *manifest) targetPath() string {
 	return fmt.Sprintf("%s%s", m.path[:len(m.path)-len(filepath.Ext(m.path))], ".yaml")
 }
 
-func (m *manifest) processTemplate(data interface{}) error {
-	if !m.template {
-		return nil
+func (m *manifest) process(fsys fs.FS, data interface{}) error {
+	manifestFile, err := fsys.Open(m.path)
+	if err != nil {
+		log.Error(err, "Failed to open manifest file", "path", m.path)
+		return err
 	}
-	path := m.targetPath()
+	defer manifestFile.Close()
 
-	f, err := os.Create(path)
+	content, err := io.ReadAll(manifestFile)
 	if err != nil {
-		log.Error(err, "Failed to create file")
-
+		log.Error(err, "Failed to read manifest file", "path", m.path)
 		return err
 	}
 
-	tmpl := template.New(m.name).Funcs(template.FuncMap{"ReplaceChar": ReplaceChar})
+	if !m.template {
+		m.processedContent = string(content)
+		return nil
+	}
```

Comment:
Perhaps a little bit of "why" can be helpful

```suggestion
	if !m.template {
		// If, by convention, the file is not suffixed with `.tmpl` we do not need to trigger template processing.
		// It's safe to return at this point.
		m.processedContent = string(content)
		return nil
	}
```